### PR TITLE
fix: fetch code from master branch instead of v2.0 in Dockerfile

### LINKED ISSUE #1145: bug(Dockerfile): use the code from master branch for building docker image
# Bug report

## Describe the bug

The `Dockerfile` in master branch seems using the code from a specific release for building docker image. I guess we should fetch the code from master branch in `Dockerfile` defined in master branch. The code from a release can only be used in a `Dockerfile` defined in corresponding release.

Any feedback is welcome. Thanks.

### PR DESCRIPTION
**Pull request by imjoey**
Signed-off-by: imjoey <majunjiev@gmail.com>

Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

Related to  #1145 .

___
### Bugfix
- Description

For now, the `Dockerfile` in master branch fetches the codes from release `v2.0` for building docker images. This would lead to uses could not get the latest code for developing or testing. IMHO, the release tag could be used here only for the Dockerfile defined in tag `v2.0` or `v2.0` branch.

- How to fix?

This PR is going to set the variable `APISIX_DASHBOARD_VERSION ` to `master`, and then we can use the latest code from master branch, as expected.

### CODE REVIEW

File: Dockerfile
Comment (juzhiyuan) [inline Dockerfile:19]: Good catch! and should we still keep this file? We have an an-in-one Dockerfile here[1]. 🤔

cc @membphis @liuxiran @gxthrj @ShiningRush 

[1] https://github.com/apache/apisix-docker/blob/master/all-in-one/apisix-dashboard/Dockerfile

Comment (imjoey) [reply Dockerfile:19]: > Good catch! and should we still keep this file? We have an an-in-one Dockerfile here[1]. 🤔
> 
> cc @membphis @liuxiran @gxthrj @ShiningRush
> 
> [1] https://github.com/apache/apisix-docker/blob/master/all-in-one/apisix-dashboard/Dockerfile

@juzhiyuan that's right. In any case, we'd better not keep two copies for a single file. The `all-in-one` is better for me (of course, we also need this patch for that). While, I would say there's one question that's worth noting: how can we build a docker image for apisix-dashboard continuously once every single commit gets merged, if using a dockerfile stored in another repository? Looking forward to your insights. Thanks.

Comment (juzhiyuan) [reply Dockerfile:19]: Oh yes, I'm working on Dashboard Image in DockerHub, once it's approved from ASF, we could add CI to auto-publish Docker Image to DockerHub.

Comment (imjoey) [reply Dockerfile:19]: > Oh yes, I'm working on Dashboard Image in DockerHub, once it's approved from ASF, we could add CI to auto-publish Docker Image to DockerHub.

@juzhiyuan that's great, and I would choose the  `all-in-one`. Let's see others' opinions. Thank you very much.

Comment (juzhiyuan) [reply Dockerfile:19]: We could merge this one first :) When that repo is created, let's reupdate this repo's documentation.

**Review by juzhiyuan — APPROVED** [review]

File: Dockerfile
Comment (imjoey) [reply Dockerfile:19]: > We could merge this one first :) When that repo is created, let's reupdate this repo's documentation.

@juzhiyuan ok, thank you very much. Shall we need to fix the same issue at https://github.com/apache/apisix-docker/blob/master/all-in-one/apisix-dashboard/Dockerfile#L36 for now?

Comment (juzhiyuan) [reply Dockerfile:19]: Ya!

Comment (imjoey) [reply Dockerfile:19]: > Ya!

@juzhiyuan ok, I'll fix that soon.

**Review by liuxiran — APPROVED** [review]

**Comment by liuxiran** [comment]
I found that this pr did not conflict with master branch, but it still needed to do `merger master`, so are there any settings? @juzhiyuan

**Comment by imjoey** [comment]
> I found that this pr did not conflict with master branch, but it still needed to do `merger master`, so are there any settings? 

@juzhiyuan @liuxiran yep. As far as I know, we need to rebase code only when there has conflicts which may completely block merging action. So shall we need keep every PR updated with the latest code base before merging? Much appreciated.

**Comment by juzhiyuan** [comment]
> I found that this pr did not conflict with master branch, but it still needed to do `merger master`, so are there any settings? @juzhiyuan

emmm Update Branch means we have to sync codes with the latest codes on the master branch, it's used to make sure every PR's action would run successfully after merged.

**Comment by liuxiran** [comment]
> > I found that this pr did not conflict with master branch, but it still needed to do `merger master`, so are there any settings? @juzhiyuan
> 
> emmm Update Branch means we have to sync codes with the latest codes on the master branch, it's used to make sure every PR's action would run successfully after merged.

got it！

**Comment by imjoey** [comment]
> > I found that this pr did not conflict with master branch, but it still needed to do `merger master`, so are there any settings? @juzhiyuan
> 
> emmm Update Branch means we have to sync codes with the latest codes on the master branch, it's used to make sure every PR's action would run successfully after merged.

ok, it's reasonable. Thanks. @juzhiyuan .

**Pull request merged**
